chalice/cli/__init__.py

```diff
@@ -175,6 +175,23 @@ def deploy_new(ctx, autogen_policy, profile, api_gateway_stage, stage):
             config.project_dir, '.chalice', 'deployed.json'))
 
 
+@cli.command('delete-new')
+@click.option('--profile', help='Override profile at deploy time.')
+@click.option('--stage', default=DEFAULT_STAGE_NAME,
+              help='Name of the Chalice stage to delete.')
+@click.pass_context
+def delete_new(ctx, profile, stage):
+    # type: (click.Context, str, str) -> None
+    factory = ctx.obj['factory']  # type: CLIFactory
+    factory.profile = profile
+    config = factory.create_config_obj(chalice_stage_name=stage)
```

Comment:
It may be worth consolidating this logic in the future with `deploy_new()`. Maybe just add a `TODO` for now. Seems to be a decent chunk of duplicated code in this function.